### ISSUE TITLE
Devise ignores (some) validations on signup

### ISSUE THREAD

**Issue by pcragone** [plataformatec/devise#4673]
tl;dr: During the signup process, I'm guessing only if trackable is turned on, Devise will call a `save(validate: false)` on the model, causing it to save despite any failed validations, due to: (https://github.com/plataformatec/devise/blob/88724e10adaf9ffd1d8dbfbaadda2b9d40de756a/lib/devise/models/trackable.rb#L33)[https://github.com/plataformatec/devise/blob/88724e10adaf9ffd1d8dbfbaadda2b9d40de756a/lib/devise/models/trackable.rb#L33]

While trying to figure out what change I had made that broke things, I discovered my application was saving with failed validations during the signup process.  In my specific case, I have an `after_create` callback that calls create on an associated model.  However, that ends up being invalid, and so does not save (which led my tests to fail).  Yet if I tried it through the Devise signup process, everything went through and that associated model saved anyway.  I eventually tracked this down to the above referenced line within devise, causing my associated model to save despite being invalid.

FWIW, this doesn't actually affect how my application ends up performing, as I need that model created despite being invalid, so it actually works in my favor (I have an after-signup profile creation process users through, so that's why the validation exists in the first place).  I don't know if this is necessarily a bug you want fixed, or just something you want documented.

**Comment by tegon** [comment]
Fixed in #4674

**Comment by carlosantoniodasilva** [comment]
Just linking a comment on the final related fix here: https://github.com/heartcombo/devise/pull/4796#issuecomment-812659563.

I'm not sure I can see how Devise would be trying to save an invalid record that wasn't already saved before.
